ssvsigner/client/client.go

```diff
@@ -0,0 +1,252 @@
+	pubKeyStrs := make([]string, 0, len(sharePubKeys))
+	for _, pubKey := range sharePubKeys {
+		pubKeyStrs = append(pubKeyStrs, hex.EncodeToString(pubKey))
+	}
+
+	req := server.RemoveValidatorRequest{
+		PublicKeys: pubKeyStrs,
+	}
+
+	reqBytes, err := json.Marshal(req)
+	if err != nil {
+		return nil, fmt.Errorf("marshal request: %w", err)
+	}
+
+	url := fmt.Sprintf("%s/v1/validators/remove", c.baseURL)
+	httpResp, err := c.httpClient.Post(url, "application/json", bytes.NewReader(reqBytes))
+	if err != nil {
+		return nil, fmt.Errorf("request failed: %w", err)
+	}
+	defer func() {
+		if err := httpResp.Body.Close(); err != nil {
+			c.logger.Error("failed to close http response body", zap.Error(err))
+		}
+	}()
+
+	respBytes, err := io.ReadAll(httpResp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("read response body: %w", err)
+	}
+
+	if httpResp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code %d: %s", httpResp.StatusCode, string(respBytes))
+	}
+
+	var resp server.RemoveValidatorResponse
+	if err := json.Unmarshal(respBytes, &resp); err != nil {
+		return nil, fmt.Errorf("unmarshal response body: %w", err)
+	}
+
+	if len(resp.Statuses) != len(sharePubKeys) {
+		return nil, fmt.Errorf("unexpected statuses length, got %d, expected %d", len(resp.Statuses), len(sharePubKeys))
+	}
+
+	return resp.Statuses, nil
+}
+
+func (c *SSVSignerClient) Sign(sharePubKey []byte, payload web3signer.SignRequest) ([]byte, error) {
+	url := fmt.Sprintf("%s/v1/validators/sign/%s", c.baseURL, hex.EncodeToString(sharePubKey))
+
+	data, err := json.Marshal(payload)
+	if err != nil {
+		return nil, fmt.Errorf("marshal request: %w", err)
+	}
+
+	resp, err := c.httpClient.Post(url, "application/json", bytes.NewReader(data))
+	if err != nil {
+		return nil, fmt.Errorf("request failed: %w", err)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			c.logger.Error("failed to close http response body", zap.Error(err))
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		respBytes, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("unexpected status code %d: %s", resp.StatusCode, string(respBytes))
+	}
+
+	signature, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("read response: %w", err)
+	}
+
+	return signature, nil
+}
+
+func (c *SSVSignerClient) GetOperatorIdentity() (string, error) {
+	url := fmt.Sprintf("%s/v1/operator/identity", c.baseURL)
+
+	resp, err := c.httpClient.Get(url)
+	if err != nil {
+		return "", fmt.Errorf("request failed: %w", err)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			c.logger.Error("failed to close http response body", zap.Error(err))
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		respBytes, _ := io.ReadAll(resp.Body)
+		return "", fmt.Errorf("unexpected status code %d: %s", resp.StatusCode, string(respBytes))
+	}
+
+	publicKey, err := io.ReadAll(resp.Body)
```

Comment:
we should avoid multiple reads of `resp.Body`

Currently, `resp.Body` is being read multiple times in some functions, which is unnecessary and can lead to issues if the logic is modified in the future. 

Suggested Fix: 
Read `resp.Body` once, store it in `respBytes`, and reuse it:

```go
respBytes, err := io.ReadAll(resp.Body)
if err != nil {
    return "", fmt.Errorf("read response: %w", err)
}

if resp.StatusCode != http.StatusOK {
    return "", fmt.Errorf("unexpected status code %d: %s", resp.StatusCode, string(respBytes))
}

return string(respBytes), nil
```

This also happens in:
`Sign`, `AddValidators`, `RemoveValidators`

Reply:
fixed